src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -108,7 +108,7 @@ public void ClosePin(int pinNumber)
             int logicalPinNumber = GetLogicalPinNumber(pinNumber);
             if (!_openPins.Contains(logicalPinNumber))
             {
-                throw new InvalidOperationException("Can not close a pin that is not open.");
+                throw new InvalidOperationException($"Can not close pin {logicalPinNumber} is not open.");
```

Comment:
I suggest to change the text to "Can not close pin {logicalPinNumber} because it is not open."